### PR TITLE
chore: move ratatui crate into workspace folder

### PR DESCRIPTION
**Pull request by joshka**
This is the first step towards modularization. Handling the move
as a separate step rather than combining it should make it easier
to rebase other PRs when necessary.

### CODE REVIEW

**Comment by orhun** [comment]
Added [`CARGO_MAKE_EXTEND_WORKSPACE_MAKEFILE`](https://sagiegurari.github.io/cargo-make/#usage-workspace-extend) environment variable to `Makefile.toml` to fix the CI

File: Makefile.toml [hunk 1]
```diff
@@ -41,6 +43,8 @@ toolchain = "nightly"
 command = "cargo"
 args = [
   "rustdoc",
+  "-p",
+  "ratatui",
```
Comment (orhun) [inline Makefile.toml:47]: There should be a way to run `rustdoc` for the workspace members (e.g. `--workspace`)

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -117,7 +117,7 @@ jobs:
       # Run cargo rustdoc with the same options that would be used by docs.rs, taking into account
       # the package.metadata.docs.rs configured in Cargo.toml.
       # https://github.com/dtolnay/cargo-docs-rs
-      - run: cargo +nightly docs-rs
+      - run: cargo +nightly docs-rs -p ratatui
```
Comment (orhun) [inline .github/workflows/ci.yml:120]: This should be run for all the workspace members.

**Review by orhun — APPROVED** [review]
LGTM.

Fixed the CI issues. There are some leftover things such as running certain cargo commands for all the workspace members but we can take a look at them in the followup PR I guess.

**Comment by joshka** [comment]
Sweet - thanks!

**Pull request merged**